alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -109,10 +109,20 @@ void AlicaContext::stepEngine()
     _engine->stepNotify();
     constexpr const auto timeout = std::chrono::seconds(2);
     auto start = std::chrono::system_clock::now();
+    int64_t current_state = 0;
+    if (_engine->getPlanBase().getDeepestNode() && _engine->getPlanBase().getDeepestNode()->getActiveState()) {
+        current_state = _engine->getPlanBase().getDeepestNode()->getActiveState()->getId();
+    }
     do {
         _engine->getAlicaClock().sleep(alica::AlicaTime::milliseconds(ALICA_LOOP_TIME_ESTIMATE));
-        if (std::chrono::system_clock::now() > start + timeout) {
-            throw std::runtime_error("Got stuck trying to step engine");
+
+        if (std::chrono::system_clock::now() > start + timeout && _stuck_in_state != current_state) {
+            _stuck_in_state = current_state;
+            Logging::logWarn(LOGNAME) << "Got stuck trying to step engine for too long in state " << current_state;
+        }
+
+        if (_stuck_in_state != current_state) {
+            _stuck_in_state = 0;
```

Comment:
variables names should be camelCase in alica

Reply:
I suppose better to reset if we get unstuck even if state name is the same, so maybe logic can be simpler